src/components/IconButton.tsx

```diff
@@ -0,0 +1,35 @@
+import cn from '@lib/utils';
+import { LucideIcon } from 'lucide-react';
+import Link from 'next/link';
+
+type Props = {
+  className: string;
+  linkURL: string;
+  categoryName: string;
+  icon: LucideIcon;
+};
+
+const IconButton = ({
+  className = '',
+  linkURL,
+  categoryName,
+  icon,
+}: Props): React.JSX.Element => {
```

Comment:
IconButton命名太廣泛，該元件似乎只有category

```suggestion
const CategoryButton: React.FC<CategoryButtonProps> = ({
  className = '',
  linkURL,
  categoryName,
  icon,
})
```

Reply:
命名修改沒問題。

但想問React.FC<Props>的type，我在React cheat上有看到，並不建議這樣使用，我看到網路上也有一些爭論，最後選擇用React.JSX.Element定義，更多參考 https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components

Reply:
那些爭議再`React 18 / typescript 5.1`都修正了，文檔和react doc上應該都有寫。

Reply:
不太認同，文件也有清楚的寫到，並不是必要的。

Reply:
如果不需要做的話，那其實不用返回，因為ts會自動推斷。
```
CategoryButton = ({  
  className = '',  
  linkURL,  
  categoryName,  
  icon,  
}: CategoryButtonProps)
```

大部分是不加返回值或加FC

Reply:
然後目前應該是沒有best practice。常見三個應該是:React.ReactNode、不加、Reac.FC。

Reply:
或許component怎麼寫可以星期一討論一下。

Reply:
那我先改成全部不加好了

Reply:
added here 037ee160952ff680829069437d7486050ff88328